tests/es6-destructuring-assignments.js

```diff
@@ -951,115 +951,6 @@ describe("ES6 destructuring assignments", () => {
         expect(scope.references[3].isWrite()).to.be.false;
     });
 
-    it("ObjectPattern in AssignmentExpression", () => {
-        const ast = espree(`
-            (function () {
```

Comment:
Maybe it would be better to replace this test with

```js
(function () {
    ({
        shorthand,
        key: value,
        hello: {
            world
        }
    } = object);
}());
```

In other words, the same as the original test, but the location of `)` second-to-last line is after `object`. That way we can still have coverage for whatever this case was testing.